### PR TITLE
Add parameter precompute_grad to CgInfluence init, adapt documentation

### LINKED ISSUE #497: Memory leak in CgInfluence
Working with a Resnet50 modek on imagenet revealed a memory issue with the implementation of [CgInfluence](https://github.com/aai-institute/pyDVL/blob/develop/src/pydvl/influence/torch/influence_function_model.py#L425).

The issue is due to precomputing the averaged gradient over the training dataset, while keeping the computation graph for computing the second order derivatives. This is not feasible for large datasets.

**Resolution**:

Expose the parameter `precopmute_grad`. Setting to False, results in the averaging of the second order derivatives over the batches, which requires less memory (no need to keep the computation graph) but is slower in general.

 The same applies to [ArnoldiInfluence](https://github.com/aai-institute/pyDVL/blob/develop/src/pydvl/influence/torch/influence_function_model.py#L717).

### PR DESCRIPTION
**Pull request by schroedk**
### Description

This PR closes #497

### Changes

- Expose the parameter `precopmute_grad`. Setting to False, results in the averaging of the second order derivatives over the batches, which requires less memory (no need to keep the computation graph) but is slower in general.

### Checklist

- [ ] ~~Wrote Unit tests (if necessary)~~
- [x] Updated Documentation (if necessary)
- [x] Updated Changelog
- [ ] ~~If notebooks were added/changed, added boilerplate cells are tagged with `"tags": ["hide"]` or `"tags": ["hide-input"]`~~

### CODE REVIEW

**Review by Xuzzo — APPROVED** [review]
Looks good to me! Thanks for the fix
